iib/workers/tasks/build.py

```diff
@@ -54,21 +61,26 @@ def _build_image(dockerfile_dir, dockerfile_name, request_id, arch):
     :param str arch: the architecture to build this image for
     :raises IIBError: if the build fails
     """
-    destination = _get_local_pull_spec(request_id, arch)
+    destination: str = _get_local_pull_spec(request_id, arch)
     log.info(
         'Building the container image with the %s dockerfile for arch %s and tagging it as %s',
         dockerfile_name,
         arch,
         destination,
     )
-    dockerfile_path = os.path.join(dockerfile_dir, dockerfile_name)
+    dockerfile_path: str = os.path.join(dockerfile_dir, dockerfile_name)
     # NOTE: It's important to provide both --override-arch and --arch to ensure the metadata
     # on the image, **and** on its config blob are set correctly.
+    #
+    # NOTE: The argument "--format docker" ensures buildah will not generate an index image with
+    # default OCI v1 manifest but always use Docker v2 format.
     run_cmd(
         [
             'buildah',
             'bud',
             '--no-cache',
+            '--format',
+            'docker',
```

Comment:
Why not set this to `v2s2`?

Reply:
According to the [manual](https://www.mankier.com/1/buildah-build#--format) the format should be either `oci` or `docker` being `oci` the default one.

Unfortunately the problem on DEV env goes beyond that... Today I tried to figure out a solution but no luck so far, but in case I don't find it do you prefer to remove this option and use this PR just for `buildah bud` retry?

Then we can investigate this issue on DEV env later, what do you think?

Reply:
Fixed.

When `buildah bud` we need to pass it as `--format docker`

When `buildah manifest ` we need to pass it as `--format v2s2`